crates/pipeline_manager/src/api/services/service_config.rs

```diff
@@ -0,0 +1,68 @@
+use crate::api::services::probe::{ServiceProbe, ServiceProbeRequest};
+use crate::api::{KafkaService, ServiceProbeResult};
+use serde::{Deserialize, Serialize};
+use std::time::Duration;
+use utoipa::ToSchema;
+
+/// Service configuration for the API
+///
+/// A Service is an API object, with as one of its properties its config.
```

Comment:
In general, if you describe the mechanics of how an API is implemented (like the enum variant + how it is stored in the DB), assume that it will drift from the actual implementation really quickly. :) Instead, describe only the contract here. Something like:

"
Configuration for a Service, which typically includes connectivity and credentials. This configuration can be used to easily derive connectors (see <some reference to the corresponding connector API>).
"

Reply:
Thanks :) Will edit the text

Reply:
Changed to:
```
Configuration for a Service, which typically includes how to establish a
connection (e.g., hostname, port) and authenticate (e.g., credentials).

This configuration can be used to easily derive connectors for the service
as well as probe it for information.
```